### PR TITLE
Chore/order claims 2

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
What:
- Begin Order Claim
- Add return items
- Claim items
- Add additional items
- Request Order Claim

Missing:
- complete http tests
- cancel claim request
- cancel claim

### CODE REVIEW

File: integration-tests/http/__tests__/claims/claims.spec.ts [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+              quantity: 1,
+            },
+          ],
+        },
+        adminHeaders
+      )
+    })
+
+    describe("Returns lifecycle", () => {
+      it.only("Full flow with 2 orders", async () => {
```
Comment (olivermrbl) [inline integration-tests/http/__tests__/claims/claims.spec.ts:387]:
```suggestion
    describe("Claims lifecycle", () => {
      it("Full flow with 2 orders", async () => {
```

File: packages/core/core-flows/src/order/steps/create-claim-items.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+export const createOrderClaimItemsStep = createStep(
+  "create-claim-items",
+  async (input: CreateOrderClaimItemsInput, { container }) => {
+    const orderModuleService = container.resolve<IOrderModuleService>(
+      ModuleRegistrationName.ORDER
+    )
+
+    const claimItems = input.changes.map((item) => {
```
Comment (olivermrbl) [inline packages/core/core-flows/src/order/steps/create-claim-items.ts:17]: question: Should we put this in a transformer in the workflow?

File: packages/core/core-flows/src/order/workflows/claim/confirm-claim-request.ts [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    orderClaim: OrderClaimDTO
+    orderChange: OrderChangeDTO
+  }) {
+    throwIfIsCancelled(order, "Order")
+    throwIfIsCancelled(orderClaim, "Claim")
+    throwIfOrderChangeIsNotActive({ orderChange })
+  }
+)
+
+function prepareFulfillmentData({
```
Comment (olivermrbl) [inline packages/core/core-flows/src/order/workflows/claim/confirm-claim-request.ts:50]: thought: Think this is 1:1 with the other `prepareFulfillmentData` – if so, we could extract to a utility

File: packages/core/core-flows/src/order/workflows/return/confirm-return-request.ts [hunk 4]
```diff
@@ -111,6 +99,31 @@ function prepareFulfillmentData({
...
+
+    for (const action of modifiedShippingMethod_.actions) {
+      if (
+        action.action === ChangeActionType.SHIPPING_ADD &&
+        action.return_id === orderReturn.id
+      ) {
+        returnShippingMethod = shippingMethod
+        break
+      }
+    }
```
Comment (olivermrbl) [inline packages/core/core-flows/src/order/workflows/return/confirm-return-request.ts:122]: question: If there is only one, we can just use `.find()` on the actions. Think that's a bit cleaner no?

File: packages/medusa/src/api/admin/claims/[id]/inbound/items/[action_id]/route.ts [hunk 5]
```diff
@@ -10,6 +10,7 @@ import {
   AuthenticatedMedusaRequest,
```
Comment (olivermrbl) [inline packages/medusa/src/api/admin/claims/[id]/inbound/items/[action_id]/route.ts:10]: question: Shouldn't we return the claim as well in all the `/claims` Rest APIs? It seems we are doing it in some but not all.

**Review by olivermrbl — APPROVED** [review]
Overall LGTM

**Comment by carlos-r-l-rodrigues** [comment]
@olivermrbl , I'm addressing you comments and a few more issues in the next PR in a few minutes.

**Pull request merged**